sdk/cosmos/azure-cosmos/README.md

```diff
@@ -588,6 +584,46 @@ def integrated_cache_snippet():
 ```
 For more information on Integrated Cache, see [Azure Cosmos DB integrated cache - Overview][cosmos_integrated_cache].
 
+### Using Transactional Batch
+Transactional batch requests allow you to send several operations to be executed at once within the same partition key.
+If all operations succeed in the order they're described within the transactional batch operation, the transaction will be committed.
+However, if any operation fails, the entire transaction is rolled back.
+
+Transactional batches have a limit of 100 operations per batch, and a total size limit of 1.2Mb for the
+batch operations being passed in.
+
+Transactional Batch operations look very similar to the singular operations apis, and are tuples containing
+(`operation_type_string`, `args_tuple`, `optional_kwargs_dictionary`):
+```python
+batch_operations = [
+        ("create", (item_body,), kwargs),
+        ("replace", (item_id, item_body), kwargs),
+        ("read", (item_id,), kwargs),
+        ("upsert", (item_body,), kwargs),
+        ("patch", (item_id, operations), kwargs),
+        ("delete", (item_id,), kwargs),
+    ]
+batch_results = container.execute_item_batch(batch_operations=batch_operations, partition_key=partition_key)
+```
+In this case, the kwargs for each operation can include `filter_predicate` in the case of using conditional patching, or
+the use of the `if_match_etag`/`if_none_match_etag` if you'd like to pass an etag filter for one single operation:
```

Comment:
We've made an effort in the SDKs to avoid exposing these names directly in the APIs - so customers are likely not familiar with them unless they've had prior reason to dig into the rest API documentation.

Reply:
Though follow up question - are these 3 options the only things supported in 3rd tuple element?
If so we could possibly get around it by documenting a typed dict:
```python
class BatchOperationOptions(TypedDict, total=False):
    filter_predicate: str
    """Used for conditional patching blah blah for example: foo"""
    if_match_etag: str
    """The etag to match against for the purpose of overwriting an existing item. If no match is found an error will be raised etc etc"""
    if_none_match_etag: str
    """The operation will only succeed if no matching etag can be found, If a match exists, the operation will fail etc etc"""
```
```

Reply:
filter_predicate only applies to patch operations, but yes this is how the same option models are defined in the Java SDK for instance:
<img width="310" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/30335873/e0add7fc-0e37-41f2-b0b2-2da09e8e189b">